### PR TITLE
SSSD documentation updated for quarkus distribution

### LINKED ISSUE #20263: SSSD documentation updated for quarkus distribution
### Description

The current SSSD documentation in the "Server admin guide" is out-of-date and it is not updated for Quarkus distribution. It should be updated for quarkus dist and make sure it works with it

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by rmartinc**
Closes https://github.com/keycloak/keycloak/issues/20263

Documentation  changes for the SSSD provider.

- The integration does not need anymore a system .so library.  It needs JDK 17 or above to work. I'm adding that note only to the upstream keycloak as downstream will only be supported with JDK-17.
- Re-adding the script that was present in wildfly dist as it's still functional.

### CODE REVIEW

File: docs/documentation/server_admin/topics/user-federation/sssd.adoc [hunk 1]
```diff
@@ -114,83 +114,64 @@ This script makes the following changes to `/etc/sssd/sssd.conf`:
...
-Install the RPM using this command:
+==== Enabling the SSSD federation provider
 
-[source,bash,subs=+attributes]
-----
-$ sudo yum install libunix-dbus-java-0.8.0-1.fc24.x86_64.rpm
-----
+{project_name} uses https://github.com/hypfvieh/dbus-java[DBus-Java] project to communicate at a low level with D-Bus and https://github.com/java-native-access/jna[JNA] to authenticate via Operating System Pluggable Authentication Modules (PAM).
 
+ifeval::[{project_community}==true]
```
Comment (mposolda) [inline docs/documentation/server_admin/topics/user-federation/sssd.adoc:159]: One minor note: Do we need this `ifeval` or can we remove it and keep the note about Java 17 always in the docs?

It seems the product would be supported just for Java 17, however in theory, you never know if something changes (or there are any support exceptions etc)? WDYT?

**Review by mposolda — COMMENTED** [review]
@rmartinc Thanks! I've added one minor comment inline.

@andymunro @douglaspalmer Are you please able to review this PR as well?

**Review by andymunro — DISMISSED** [review]

File: docs/documentation/server_admin/topics/user-federation/sssd.adoc (hunk 1 above)
Comment (rmartinc) [reply docs/documentation/server_admin/topics/user-federation/sssd.adoc]: Yep, it's OK for me. I just differentiate between community and product to not mention jdk-17 in product if jdk-17 was compulsory. It's done now.

**Review by mposolda — APPROVED** [review]
@rmartinc @andymunro Thanks for the updates and review!

**Pull request merged**